tests/integration/Render/RenderTest.php

```diff
@@ -0,0 +1,146 @@
+						'type' => 'string',
+					],
+				],
+			],
+		] );
+
+		$registration_result = register_remote_data_block( [
+			'title' => 'Test ZIP API',
+			'render_query' => [
+				'query' => $test_query,
+			],
+		] );
+
+		$this->assertTrue( $registration_result );
+
+		// Register block configuration with WordPress, normally done during the 'init' filter
+		$block_config = ConfigStore::get_block_configuration( 'remote-data-blocks/test-zip-api' );
+		$this->assertTrue( is_array( $block_config ) && [] !== $block_config );
+		BlockRegistration::register_block_configuration( $block_config );
+
+		$result_html = do_blocks('
+			<!-- wp:remote-data-blocks/test-zip-api {"remoteData":{"blockName":"remote-data-blocks/test-zip-api","queryInput":{"zip_code":"12345"}}} -->
+			<div class="wp-block-remote-data-blocks-test-zip-api rdb-container">
+				<!-- wp:heading {"metadata":{"bindings":{"content":{"source":"remote-data/binding","args":{"block":"remote-data-blocks/test-zip-api","field":"zip_code"}}},"name":"Zip Code"}} -->
+				<h2 id="field-zip-code" class="wp-block-heading"></h2>
+				<!-- /wp:heading -->
+
+				<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"remote-data/binding","args":{"block":"remote-data-blocks/test-zip-api","field":"city"}}},"name":"City"}} -->
+				<p id="field-city"></p>
+				<!-- /wp:paragraph -->
+
+				<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"remote-data/binding","args":{"block":"remote-data-blocks/test-zip-api","field":"state"}}},"name":"State"}} -->
+				<p id="field-state"></p>
+				<!-- /wp:paragraph -->
+			</div>
+			<!-- /wp:remote-data-blocks/test-zip-api -->
+		');
+
+		$dom = self::load_html_into_dom( $result_html );
+
+		$this->assertIdInDomHasContent( $dom, 'field-zip-code', '12345' );
+		$this->assertIdInDomHasContent( $dom, 'field-city', 'Test City' );
+		$this->assertIdInDomHasContent( $dom, 'field-state', 'Test State' );
+	}
+
+	private function get_query_runner_with_response( array $response_data ) {
+		return new class($response_data) extends QueryRunner {
+			private $response_data;
+
+			public function __construct( $response_data ) {
+				$this->response_data = $response_data;
+			}
+
+			protected function get_raw_response_data( HttpQueryInterface $query, array $input_variables ): array|WP_Error {
+				return [
+					'metadata' => [
+						'age' => 100,
+						'status_code' => 200,
+					],
+					'response_data' => $this->response_data,
+				];
+			}
+		};
+	}
+
+	private function load_html_into_dom( $html ): DOMDocument {
+		$dom = new DOMDocument();
+		$dom_html = '<?xml encoding="UTF-8">' . $html;
+
+		try {
+			$dom->loadHTML( $html, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
+		} catch ( \Throwable $e ) {
+			$this->fail( sprintf( 'Failed to parse rendered block as HTML: %s', $e ) );
+		}
+
+		return $dom;
+	}
+
+	private function assertIdInDomHasContent( DOMDocument $dom, string $html_id, string $expected_content ): void {
```

Comment:
It will probably make sense to move these private functions to a helper class as WP-PHPUnit tests expand, but I've kept them in here until necessary.